docs/configuring_previous_line_rules.rst

```diff
@@ -0,0 +1,133 @@
+Configuring Previous Line Rules
+-------------------------------
+
+There are rules which will check the contents on lines above code structures.
+These rules allow enforcement of comments and blank lines.
+
+There are several options to these rules, which can be selected by using the :code:`method` option:
+
++---------------------+--------------------------------------------------+
+| Method              | Description                                      |
++=====================+==================================================+
+| require_blank_line  | Requires a blank line on the line above.         |
++---------------------+--------------------------------------------------+
+| no_code             | Either a blank or comment on the line above.     |
++---------------------+--------------------------------------------------+
+| allow_comment       | Either a blank or comment on line above, with a  |
+|                     | blank line above the comment.                    |
++---------------------+--------------------------------------------------+
+| require_comment     | Requires a comment on the line above, with a     |
+|                     | blank line above the comment.                    |
++---------------------+--------------------------------------------------+
+
+This is an example of how to configure these options.
+
+.. code-block:: yaml
+
+   rule :
+     entity_003:
+        method : require_blank_line
+
+.. NOTE:: All examples below are using the rule **entity_004**.
+
+Example: require_blank_line
+###########################
+
+The following code would fail with this option:
```

Comment:
I think you should always give the fail example first, then the pass.

This is in alignment with the general rule violations: you first always show the violation and then the fix.

Reply:
hm ... ok, the difference here is that it's not a fix that you suggest, right? ...

Well, in that case, simply make the examples have pass and fail in the same order :wink:

Reply:
I thought about making it a fix, but thought just illustrating what would trigger and what would not trigger the rule would be  clearer.